lib/msf/core/post/linux/system.rb

```diff
@@ -102,6 +102,216 @@ def get_sysinfo
   end
 
+  #
+  # Returns all listening services along with their ports
+  # @param portsonly Return the listening ports without their associated service
+  # @return [Hash]
+  #
+  def get_listening_services(portsonly = false)
+    services = {}
+    begin
+      full = cmd_exec('netstat -tulpn').to_s
+      raise "You must be root to get listening ports" if full.include? '(No info could be read'
+      full = full.split("\n")[2..-1]
+      full.delete!(':') # Only happens when getting services
+
+      if portsonly
+        ports = []
+        full.each do |p|
+          ports << p.split('/')[1]
+        end
+        return ports
+      else
+        full.each do |s|
+          split = s.split('/')
+          services[:"#{split[0]}"] = split[1]
+        end
+      end
+
+      services
+    rescue
+      raise 'Could not gather listening ports'
+    end
+  end
+
+  #
+  # Gathers all SUID files on the filesystem.
+  # NOTE: This uses the Linux `find` command. It will most likely take a while to get all files.
+  # Consider specifying a more narrow find path.
+  # @param findpath The path on the system to start searching
+  # @return [Array]
+  def get_suid_files(findpath = '/')
+    cmd_exec("find #{findpath} -perm -4000 -print -xdev").to_s.split("\n")
+  rescue
+    raise "Could not retrieve all SUID files"
+  end
+
+  #
+  # Gets the $PATH environment variable
+  #
+  def get_path
+    cmd_exec('echo $PATH').to_s
+  rescue
+    raise "Unable to determine path"
+  end
+
+  #
+  # Gets basic information about the system's CPU.
+  # @return [Hash]
+  #
+  def get_cpu_info
+    info = {}
+    orig = cmd_exec("cat /proc/cpuinfo").to_s
+    cpuinfo = orig.split("\n\n")[0]
+    # This is probably a more platform independent way to parse the results (compared to splitting and assigning preset indices to values)
+    cpuinfo.split("\n").each do |l|
+      info[:speed_mhz]   = l.split(': ')[1].to_i if l.include? 'cpu MHz'
+      info[:product]     = l.split(': ')[1]      if l.include? 'model name'
+      info[:vendor]      = l.split(': ')[1]      if l.include? 'vendor_id'
+    end
+    info[:cores] = orig.split("\n\n").size
+    info
+  rescue
+    raise "Could not get CPU information"
+  end
+
+  #
+  # Gets the hostname of the system
+  # @return [String]
+  #
+  def get_hostname
+    cmd_exec('uname -n').to_s
+  rescue
+    raise 'Unable to retrieve hostname'
+  end
+
+  #
+  # Gets the name of the current shell
+  # @return [String]
+  #
+  def get_shell_name
+    psout = cmd_exec('ps -p $$').to_s
+    psout.split("\n").last.split(' ')[3]
+  rescue
+    raise 'Unable to gather shell name'
+  end
+
+  #
+  # Checks if the system has gcc installed
+  # @return [Boolean]
+  #
+  def has_gcc?
+    command_exists? 'gcc'
+  rescue
+    raise 'Unable to check for gcc'
+  end
+
+  #
+  # Gets the version of gcc
+  # @return [String]
+  #
+  def gcc_version
+    return cmd_exec('gcc --version | head -n 1').to_s.split(' ')[3] if has_gcc?
```

Comment:
Seems this match has some problems.

gcc on Fedora version string:

```
gcc (GCC) 6.2.1 20160916 (Red Hat 6.2.1-2)
```

As opposed to Ubuntu:

```
gcc (Ubuntu 4.8.4-2ubuntu1~14.04.4) 4.8.4
```